src/zope/i18n/tests/test_formats.py

```diff
@@ -77,22 +90,25 @@ def getDayNames(self):
         return [self.days.get(type, (None, None))[0] for type in range(1, 8)]
 
     def getDayTypeFromName(self, name):
-        for item in self.days.items():
-            if item[1][0] == name:
-                return item[0]
+        raise NotImplementedError()
 
     def getDayAbbreviations(self):
         return [self.days.get(type, (None, None))[1] for type in range(1, 8)]
 
     def getDayTypeFromAbbreviation(self, abbr):
-        for item in self.days.items():
-            if item[1][1] == abbr:
-                return item[0]
+        raise NotImplementedError()
+
 
+class _TestCase(TestCase):
+    # Avoid deprecation warnings in Python 3 by making the preferred
+    # method name available for Python 2.
+    assertRaisesRegex = getattr(TestCase, 'assertRaisesRegex', TestCase.assertRaisesRegexp)
```

Comment:
Nice trick!

Reply:
It's amazing what you can do in one unconditional line.